### PR TITLE
Add PlatformIO to CI

### LINKED ISSUE #1110: [Bug]: Doesn't compile for Teensy 4.1
### The type of board you are using.

Teensy board

### The boards name or FQBN (e.g. esp8266:esp8266:d1_mini:eesz=4M3M,xtal=80) for unspecified board.

Teensy 4.1

### The board manager URL used for unspecified board.

See ArduinoTeensy

### What IDE are you using?

PlatformIO IDE

### What IR protocol are you using?

Unknown

### Pin(s) used for IR-receive, if not default.

Arduino pin number: default

### Example(s) you have checked while hunting the bug.

- [X] SimpleReceiver
- [ ] ReceiveDemo
- [ ] SendRawDemo
- [ ] ReceiverTimingAnalysis
- [ ] TinyReceiver
- [ ] TinySender
- [ ] ReceiveAndSend
- [ ] SimpleSender
- [ ] SendDemo
- [ ] SendLGAirConditionerDemo
- [ ] UnitTest
- [ ] Other - please specify below
- [X] I checked, if at least one of the examples was working.

### Example(s) to reproduce the issue.

- [X] SimpleReceiver
- [ ] ReceiveDemo
- [ ] SendRawDemo
- [ ] ReceiverTimingAnalysis
- [ ] TinyReceiver
- [ ] TinySender
- [ ] ReceiveAndSend
- [ ] SimpleSender
- [ ] SendDemo
- [ ] SendLGAirConditionerDemo
- [ ] UnitTest
- [ ] Other - please specify below

### The library version you are working with.

- [X] I use the [latest Arduino library version](../releases) and verified this!
- [X] I use the latest repo version ([download link](../archive/master.zip)) and verified this!

### What are the steps to reproduce this issue?

1. Create new PlatformIO project for Teensy 4.1
2. add library
```ini
[env:teensy41]
platform = teensy
board = teensy41
framework = arduino
lib_deps = https://github.com/Arduino-IRremote/Arduino-IRremote.git
```
3. Copy example files from https://github.com/Arduino-IRremote/Arduino-IRremote/tree/master/examples/SimpleReceiver in `src/`
4. Hit "Build"

### What happens?

Fails to compile.
```
In file included from .pio/libdeps/teensy41/IRremote/src/IRremote.hpp:274:0,
                 from src/main.cpp:67:
.pio/libdeps/teensy41/IRremote/src/private/IRTimer.hpp: In function 'void timerEnableReceiveInterrupt()':
.pio/libdeps/teensy41/IRremote/src/private/IRTimer.hpp:1240:43: error: 'pwm1_3_isr' was not declared in this scope
     attachInterruptVector(IRQ_FLEXPWM1_3, pwm1_3_isr);
                                           ^
```

### The serial output which indicates the error happened.

```shell
None relevant
```

### What were you expecting to happen?

Compilation os OK.

### Additional context.

Although https://github.com/Arduino-IRremote/Arduino-IRremote/pull/928 was merged, this fails to compile.

### Final checklist for the bug report.

- [X] I have **read** the README.md file thoroughly
- [X] I have searched existing issues to see if there is anything I have missed.
- [X] I have browsed the examples for one, that matches my use case.
- [X] The title of the issue is helpful and relevant.

### PR DESCRIPTION
**Pull request by maxgerhardt**
* Adds PlatformIO CI for nearly every example but the ATTiny ones for 6 boards by using `pio ci --lib="." --board=uno --board=featheresp32 --board=nodemcuv2 --board=bluepill_f103c8 --board=pico --board=teensy41`
* Thus adds previously missing Teensy CI
* Prepares the Github action file to also use the regular Teensy board package
   * fails to compile because arduino-cli prefers the 3.x IRemote library built into the Teensy core :( see https://github.com/arduino/arduino-cli/issues/2106#issuecomment-1470799272
* Fixes #1110 by forward declaring the ISR function
  * the Teensy-core builtin library versions did that the same way, but some code additions and missing CI made this line be at the wrong place.

Runs see https://github.com/maxgerhardt/Arduino-IRremote/actions/runs/4430693134

### CODE REVIEW

File: src/private/IRTimer.hpp [hunk 1]
```diff
@@ -1498,7 +1500,7 @@ void timerConfigForReceive() {
         ; // wait for sync to ensure that we can write again to COUNT16.CTRLA.reg
     // Reset TCx
     TC->CTRLA.reg = TC_CTRLA_SWRST;
-    // When writing a �1� to the CTRLA.SWRST bit it will immediately read as �1�.
```
Comment (ArminJo) [inline src/private/IRTimer.hpp:1501]: Thanks!

**Comment by ArminJo** [comment]
Thanks Max,
this is a good starting point!
- Is it OK if I commit the changes to IRTimer.hpp by myself, just to fix this error first, or can I pull only one file from a pull request or want you to create a PR with yiust this file?
- Can you put the PlatformIO CI into a separate file LibraryBuildPlatformIO.yml?
- For me, your CI produces to many jobs, normally I see one job for all examples. If it produces only one job and the examples are specified by wildcards or an exclusion list, it is way more versatile and I can use it for my other ten libraries 😀.

**Comment by maxgerhardt** [comment]
The current CI shoots off 28 jobs, see https://github.com/Arduino-IRremote/Arduino-IRremote/actions/runs/4421562696. That is all examples **per-board**.

My PlatformIO works per-example, thus only adding 22.

And sure I can refactor it into a new workflow. And do the Teensy fix as you please, then this can become a CI-only PR.

**Comment by ArminJo** [comment]
Yes you are right,
but what happens If I want to check the other platforms also with PlatformIO ;-).

**Comment by ArminJo** [comment]
Hi Mark,
can you provide an action, which is more generic and not "mundgemalt" for every example? It would be awesome!
Thanks
Armin

**Comment by maxgerhardt** [comment]
Sure I'll try to refactor it into its own .yml action file and do automatic example discovery with an optional exclude list.

**Comment by ArminJo** [comment]
Perfect!!! 🥇

**Comment by ArminJo** [comment]
Any progress?

**Pull request closed without merging**